lib/src/components/NxToast/__tests__/NxToastContainer.test.tsx

```diff
@@ -0,0 +1,161 @@
+      async function() {
+
+        const user = userEvent.setup();
+
+        const MountingBehavior = () => {
+          const [showToast, toggleToast] = useToggle(false);
+          return (
+            <>
+              <button type="button" onClick={()=>toggleToast()}>Show Toast</button>
+              <NxToastContainer>
+                {showToast &&
+                <NxToast onClose={()=>toggleToast()}>
+                  <NxErrorAlert>message</NxErrorAlert>
+                </NxToast>
+                }
+              </NxToastContainer>
+            </>
+          );
+        };
+        const { container } = render(<MountingBehavior/>);
+
+        expect(screen.queryByRole('alert')).not.toBeInTheDocument();
+
+        const button = screen.getByRole('button', {name: 'Show Toast'});
+        await user.click(button);
+
+        expect(screen.getByRole('alert')).toBeVisible();
+
+        const toast = container.querySelector('.nx-toast');
+        const closeBtn = screen.getByRole('button', {name: 'Close'});
+        await user.click(closeBtn);
+        fireEvent.animationEnd(toast as Element);
+
+        expect(toast).not.toBeInTheDocument();
+      });
+
+  it('tests focusing when NxToasts mount and unmount', async function() {
+    const user = userEvent.setup();
+
+    interface ToastModel {
+      id: number;
+    }
+
+    const FocusBehavior = () => {
+      const [toastIdInc, setToastIdInc] = useState<number>(0);
+      const [toasts, setToasts] = useState<ToastModel[]>([]);
+
+      const addToast = () => {
+        const toastId = toastIdInc + 1;
+        setToastIdInc(toastId);
+        setToasts([
+          { id: toastId },
+          ...toasts
+        ]);
+      };
+      const removeToast = (id: number) => setToasts(reject(propEq('id', id), toasts));
+      return (
+        <>
+          <button type="button" onClick={()=>addToast()}>Show Toast</button>
+          <NxToastContainer>
+            {
+              toasts.map(({id}) => {
+                return (
+                  <NxToast key={id}
+                           onClose={()=> removeToast(id)}>
+                    <NxErrorAlert>Error Message</NxErrorAlert>
+                  </NxToast>
+                );
+              })
+            }
+          </NxToastContainer>
+        </>
+      );
+    };
+    const {container} = render(<FocusBehavior />);
+
+    expect(screen.queryByRole('alert')).not.toBeInTheDocument();
+
+    const button = screen.getByRole('button', {name: 'Show Toast'});
+    await user.click(button);
+    await user.click(button);
+    await user.click(button);
+
+    // use let because both arrays will be re-defined multiple times as btns are added / removed
+    let closeBtns = await screen.findAllByRole('button', {name: 'Close'});
+    let toasts = container.querySelectorAll('.nx-toast');
+
+    expect(toasts).toHaveLength(3);
+    // topmost close button has focus before any toasts are closed
+    expect(closeBtns[0]).toHaveFocus();
```

Comment:
As I mentioned in another comment, testing that a newly-opened toast's close button has focus should be done in the `NxToast` tests

Reply:
> As I mentioned in another comment, testing that a newly-opened toast's close button has focus should be done in the `NxToast` tests


The test inside NxToast was implemented here: [3551f0c](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/3551f0ce20a93aa7807c4432b75f9adff6c97ad0)
The test inside NxToastContainer was removed here: [124d31a](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/124d31a171f3c37e46b17f1a6acb9e74d964e45d), and NxToast focus test was updated.